napari/__main__.py

```diff
@@ -2,32 +2,89 @@
 napari command line viewer.
 """
 import argparse
+import logging
 import sys
+from ast import literal_eval
+from typing import Any, Dict, List
 
 from . import __version__, gui_qt, view_path
 from .utils import citation_text, sys_info
 
 
 class InfoAction(argparse.Action):
     def __call__(self, *args, **kwargs):
+        # prevent unrelated INFO logs when doing "napari --info"
+        logging.basicConfig(level=logging.WARNING)
         print(sys_info())
         sys.exit()
 
 
 class CitationAction(argparse.Action):
     def __call__(self, *args, **kwargs):
+        # prevent unrelated INFO logs when doing "napari --citation"
+        logging.basicConfig(level=logging.WARNING)
         print(citation_text)
         sys.exit()
 
 
+def validate_unknown_args(unknown: List[str]) -> Dict[str, Any]:
+    """Convert a list of strings into a dict of valid kwargs for add_* methods.
+
+    Will exit program if any of the arguments are unrecognized, or are
+    malformed.  Converts string to python type using literal_eval.
+
+    Parameters
+    ----------
+    unknown : List[str]
+        a list of strings gathered as "unknown" arguments in argparse.
+
+    Returns
+    -------
+    kwargs : Dict[str, Any]
+        {key: val} dict suitable for the viewer.add_* methods where ``val``
+        is a ``literal_eval`` result, or string.
+    """
+    from .components.add_layers_mixin import valid_add_kwargs
+
+    out: Dict[str, Any] = dict()
+    valid = set.union(*valid_add_kwargs().values())
+    for i, arg in enumerate(unknown):
+        if not arg.startswith("--"):
+            continue
+        if "=" in arg:
+            sys.exit(f"error: '=' in argument {arg}. (Use space instead)")
+        key = arg[2:].replace("-", "_")
```

Comment:
2?

Reply:
trimming the "--" prefix, which we asserted was present a couple lines above

Reply:
Ah! :facepalm: 

Ok, this is nitpicky, but since I want you to fix the `=` anyway: how about:

```suggestion
        key = arg.lstrip('-').replace("-", "_")
```